src/sql/base/browser/ui/table/tableCellEditorFactory.ts

```diff
@@ -174,4 +175,76 @@ export class TableCellEditorFactory {
 		}
 		return TextEditor;
 	}
+
+	public getDropDownEditorClass(context: any, defaultOptions: string[]): any {
+		const self = this;
+		class TextEditor {
+			private _originalValue: string;
+			private _dropdown: Dropdown;
+			private _keyCaptureList: number[];
+
+			constructor(private _args: Slick.Editors.EditorOptions<Slick.SlickData>) {
+				this.init();
+				const keycodesToCapture = [KeyCode.Home, KeyCode.End, KeyCode.UpArrow, KeyCode.DownArrow, KeyCode.LeftArrow, KeyCode.RightArrow];
+				this._keyCaptureList = keycodesToCapture.map(keycode => getCodeForKeyCode(keycode));
+			}
+
+			/**
+			 * The text editor should handle these key press events to avoid event bubble up
+			 */
+			public get keyCaptureList(): number[] {
+				return this._keyCaptureList;
+			}
+
+			public init(): void {
+				const container = DOM.$('');
+				this._args.container.appendChild(container);
+				this._dropdown = new Dropdown(container, self._contextViewProvider);
+				container.style.height = '100%';
```

Comment:
Following previous implementations as Alan suggested.

Reply:
@alanrenmsft Any reason not to have these in a css?

Reply:
one benefit of this approach is that you know the style will work for sure since it is applied at the element directly. I don't have a problem moving this to css file with very specific selectors. I don't favor one approach over the other if we only need to set a couple styles.